### PR TITLE
Kotlinify codebase

### PR DESCRIPTION
**Pull request by SimonMarquis**
- Remove unnecessary nullable types
- Replace no-op  method bodies with Unit
- Convert to expression body
- Replace if with when
- Remove braces from 'when' entries
- Remove braces from if statement
- Convert to single line lambda
- oneline if/returns
- Replace 'contains' call with 'in' operator

Following this refactor, it could be great to envision a more "strict" code formatter like ktlint 1.0 (we are currently stuck at 0.48.1)

### CODE REVIEW

File: core/designsystem/src/main/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/component/scrollbar/ScrollbarExt.kt [hunk 1]
```diff
@@ -230,10 +230,5 @@ fun LazyStaggeredGridState.scrollbarState(
...
 
-private inline fun <T> List<T>.floatSumOf(selector: (T) -> Float): Float {
-    var sum = 0f
-    for (element in this) {
-        sum += selector(element)
-    }
-    return sum
-}
+private inline fun <T> List<T>.floatSumOf(selector: (T) -> Float): Float =
+    fold(0f) { acc, it -> acc + selector(it) }
```
Comment (dturner) [inline core/designsystem/src/main/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/component/scrollbar/ScrollbarExt.kt:234]: Just trying to make this clearer for a first time reader
```suggestion
    fold(initial = 0f) { accumulator, listItem -> accumulator + selector(listItem) }
```

File: feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/navigation/BookmarksNavigation.kt [hunk 2]
```diff
@@ -24,7 +24,7 @@ import com.google.samples.apps.nowinandroid.feature.bookmarks.BookmarksRoute
 
 const val BOOKMARKS_ROUTE = "bookmarks_route"
 
-fun NavController.navigateToBookmarks(navOptions: NavOptions? = null) {
+fun NavController.navigateToBookmarks(navOptions: NavOptions) {
     this.navigate(BOOKMARKS_ROUTE, navOptions)
```
Comment (dturner) [inline feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/navigation/BookmarksNavigation.kt:28]: brackets could also be removed so becomes

```
fun NavController.navigateToBookmarks(navOptions: NavOptions) =
    this.navigate(BOOKMARKS_ROUTE, navOptions)
```

File: feature/foryou/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/foryou/navigation/ForYouNavigation.kt [hunk 3]
```diff
@@ -30,7 +30,7 @@ const val FOR_YOU_ROUTE = "for_you_route/{$LINKED_NEWS_RESOURCE_ID}"
 private const val DEEP_LINK_URI_PATTERN =
     "https://www.nowinandroid.apps.samples.google.com/foryou/{$LINKED_NEWS_RESOURCE_ID}"
 
-fun NavController.navigateToForYou(navOptions: NavOptions? = null) {
+fun NavController.navigateToForYou(navOptions: NavOptions) {
```
Comment (dturner) [inline feature/foryou/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/foryou/navigation/ForYouNavigation.kt:33]: Brackets can be removed

File: feature/interests/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/interests/navigation/InterestsNavigation.kt [hunk 4]
```diff
@@ -26,7 +26,7 @@ import com.google.samples.apps.nowinandroid.feature.interests.InterestsRoute
 private const val INTERESTS_GRAPH_ROUTE_PATTERN = "interests_graph"
 const val INTERESTS_ROUTE = "interests_route"
 
-fun NavController.navigateToInterestsGraph(navOptions: NavOptions? = null) {
+fun NavController.navigateToInterestsGraph(navOptions: NavOptions) {
```
Comment (dturner) [inline feature/interests/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/interests/navigation/InterestsNavigation.kt:29]: Brackets can be removed

File: feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/navigation/BookmarksNavigation.kt (hunk 2 above)
Comment (dturner) [reply feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/navigation/BookmarksNavigation.kt:28]: Also `this.` can be removed

File: feature/foryou/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/foryou/navigation/ForYouNavigation.kt [hunk 5]
```diff
@@ -30,7 +30,7 @@ const val FOR_YOU_ROUTE = "for_you_route/{$LINKED_NEWS_RESOURCE_ID}"
...
     this.navigate(FOR_YOU_ROUTE, navOptions)
```
Comment (dturner) [inline feature/foryou/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/foryou/navigation/ForYouNavigation.kt:34]: Also `this.` can be removed

File: feature/interests/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/interests/navigation/InterestsNavigation.kt [hunk 6]
```diff
@@ -26,7 +26,7 @@ import com.google.samples.apps.nowinandroid.feature.interests.InterestsRoute
...
     this.navigate(INTERESTS_GRAPH_ROUTE_PATTERN, navOptions)
```
Comment (dturner) [inline feature/interests/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/interests/navigation/InterestsNavigation.kt:30]: Also `this.` can be removed

**Review by dturner — CHANGES_REQUESTED** [review]
Thanks very much for doing this. Idiomatic Kotlin is definitely nicer to read.

**Comment by dturner** [comment]
> it could be great to envision a more "strict" code formatter like ktlint 1.0 (we are currently stuck at 0.48.1)

ktlint was merged in #1094

**Comment by SimonMarquis** [comment]
@dturner PTAL, all your comments have been addressed 🤓

**Review by dturner — APPROVED** [review]

**Comment by dturner** [comment]
@SimonMarquis I can't see any changes relating to the resolved changes - did you forget to push a commit?

**Review by dturner — CHANGES_REQUESTED** [review]
Waiting on suggested changes to be addressed

**Comment by SimonMarquis** [comment]
@dturner these commits are:
- https://github.com/android/nowinandroid/pull/1039/commits/bdb10a687079ca0fd684fcbd31f729c84a9cfc51
- https://github.com/android/nowinandroid/pull/1039/commits/65f0d5b2d9cb1a743e91dbd7eb06423baf9c03a7
- https://github.com/android/nowinandroid/pull/1039/commits/6cd5958665f6c54aeb082a4acac28967b6dc946a
- https://github.com/android/nowinandroid/pull/1039/commits/b7a628ece5852dadbd4e2c69711ebbe24bcf45fc
- https://github.com/android/nowinandroid/pull/1039/commits/cd62fd5e01ddf8902dd5c86bd2f0a3b368f8e34f

**Comment by dturner** [comment]
Oh apologies @SimonMarquis, for some reason I was reviewing outdated files which didn't have those commits.

**Review by dturner — APPROVED** [review]

**Pull request merged**